playbooks/ceph.yml

```diff
@@ -368,7 +368,7 @@
         cifmw_cephadm_dashboard_key: "{{ cifmw_cephadm_key }}"
 
     - name: Create cephfs volume
-      when: cifmw_ceph_daemons_layout.cephfs_enabled  | default(true) | bool
+      when: (cifmw_ceph_daemons_layout.cephfs_enabled | default(true) | bool) or (cifmw_ceph_daemons_layout.ceph_nfs_enabled | default(true) | bool)
```

Comment:
Thinking better about this, we should split cephfs and cephnfs, as requested.
We can leave the current code as it is, and add something like:

```
   when: (cifmw_ceph_daemons_layout.cephnfs_enabled | default(true) | bool)
   ansible.builtin.import_role:
        name: cifmw_cephadm
        tasks_from: cephnfs
      vars:
        # we reuse the same VIP reserved for rgw
        cifmw_cephadm_nfs_vip: "{{ cifmw_cephadm_vip }}/24"
```

Then, within  `cephnfs`, you can `include_tasks: cephfs.yaml` to get mds deployed even in case of NFS.

Reply:
Agree with you.. i will change it in the next patch...
we should ensure that cephfs.yaml tasks are not executed twice.

Reply:
ack and thanks, as long as @tosky is ok with that we can merge. 
BTW from a deployment perspective scenario4 is valid and consistent with a ctlplane where manila doesn't have a share configured w/ native_cephfs but only ceph_nfs.
I think the flow is correct and we can move forward with this.